frontend/ui/widget/readerprogress.lua

```diff
@@ -0,0 +1,481 @@
+local InputContainer = require("ui/widget/container/inputcontainer")
+local FrameContainer = require("ui/widget/container/framecontainer")
+local CenterContainer = require("ui/widget/container/centercontainer")
+local LeftContainer = require("ui/widget/container/leftcontainer")
+local HorizontalGroup = require("ui/widget/horizontalgroup")
+local OverlapGroup = require("ui/widget/overlapgroup")
+local VerticalGroup = require("ui/widget/verticalgroup")
+local HorizontalSpan = require("ui/widget/horizontalspan")
+local VerticalSpan = require("ui/widget/verticalspan")
+local ProgressWidget = require("ui/widget/progresswidget")
+local LineWidget = require("ui/widget/linewidget")
+local TextWidget = require("ui/widget/textwidget")
+local CloseButton = require("ui/widget/closebutton")
+local Geom = require("ui/geometry")
+local Blitbuffer = require("ffi/blitbuffer")
+local Screen = require("device").screen
+local Font = require("ui/font")
+local _ = require("gettext")
+local UIManager = require("ui/uimanager")
+local util = require("util")
+
+local progress_dates = {}
+
+local ReaderProgress = InputContainer:new{
+    padding = Screen:scaleBySize(15),
+}
+
+function ReaderProgress:init()
+    self.small_font_face = Font:getFace("ffont", 15)
+    self.medium_font_face = Font:getFace("ffont", 20)
+    self.large_font_face = Font:getFace("ffont", 25)
+    progress_dates = self.dates
```

Comment:
When I try using self.dates e.g. in function getTotalStats()
```

function ReaderProgress:getTotalStats()
    local total_time = 0
    local total_pages = 0
    for _, v in pairs(self.dates) do
        total_pages = total_pages + v. count
        total_time = total_time + v.read
    end
    return total_time, total_pages
end
```
and use it in function ReaderProgress:genSummaryWeek(width) like that:
`
    local total_time, total_pages = self.getTotalStats()`

I get an error
```

./luajit: plugins/statistics.koplugin/readerprogress.lua:52: attempt to index local 'self' (a nil value)
stack traceback:
	plugins/statistics.koplugin/readerprogress.lua:52: in function 'getTotalStats'
	plugins/statistics.koplugin/readerprogress.lua:344: in function 'genSummaryWeek'
	plugins/statistics.koplugin/readerprogress.lua:78: in function 'getStatusContent'
	plugins/statistics.koplugin/readerprogress.lua:45: in function 'init'
	frontend/ui/widget/widget.lua:39: in function 'new'
	plugins/statistics.koplugin/main.lua:184: in function 'callback'
	frontend/ui/widget/touchmenu.lua:567: in function 'action'
	frontend/ui/uimanager.lua:466: in function '_checkTasks'
	frontend/ui/uimanager.lua:612: in function 'handleInput'
	frontend/ui/uimanager.lua:706: in function 'run'
	./reader.lua:188: in main chunk
	[C]: at 0x00404750

```

self.dates has a nil value.